Update mapping dictionary to support functionalmodules and pooling operations

raghuramank100 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #24924 Serialization for nn.quantized.functional modules
* #24923 Support observer without any data calibration
* **#24804 Update mapping dictionary to support functionalmodules and pooling operations**
* #24803 Handle empty qconfig for functional Modules
* #24789 Work around for bias quantization for conv and linear operators

Differential Revision: [D16879132](https://our.internmc.facebook.com/intern/diff/D16879132/)

raghuramank10000 committed:
Update mapping dictionary to support functionalmodules and pooling operations



Differential Revision: [D16879132](https://our.internmc.facebook.com/intern/diff/D16879132/)

raghuramank10000 committed:
Update on "Update mapping dictionary to support functionalmodules and pooling operations"


Differential Revision: [D16879132](https://our.internmc.facebook.com/intern/diff/D16879132/)

raghuramank10000 committed:
Update on "Update mapping dictionary to support functionalmodules and pooling operations"


Differential Revision: [D16879132](https://our.internmc.facebook.com/intern/diff/D16879132/)

jerryzh168 commented on torch/quantization/quantize.py:
```diff
@@ -199,6 +199,7 @@ def forward(self, x):
     nn.Conv2d: nnq.Conv2d,
     QuantStub: nnq.Quantize,
     DeQuantStub: nnq.DeQuantize,
+    nn.quantized.FloatFunctional: nn.quantized.QFunctional,
```
nit: nn.quantized -> nnq

jerryzh168 commented on test/test_quantization.py:
```diff
@@ -428,6 +450,7 @@ def test_manual(self):
         test_only_train_fn(model, self.train_data)
         convert(model)
 
+
```
remove?

jerryzh168 reviewed (approved):
LGTM

raghuramank10000 committed:
Update on "Update mapping dictionary to support functionalmodules and pooling operations"


Differential Revision: [D16879132](https://our.internmc.facebook.com/intern/diff/D16879132/)

raghuramank10000 committed:
Update on "Update mapping dictionary to support functionalmodules and pooling operations"


Differential Revision: [D16879132](https://our.internmc.facebook.com/intern/diff/D16879132/)

raghuramank10000 committed:
Update on "Update mapping dictionary to support functionalmodules and pooling operations"


Differential Revision: [D16879132](https://our.internmc.facebook.com/intern/diff/D16879132/)

raghuramank10000 committed:
Update on "Update mapping dictionary to support functionalmodules and pooling operations"


Differential Revision: [D16879132](https://our.internmc.facebook.com/intern/diff/D16879132/)

raghuramank10000 committed:
Update on "Update mapping dictionary to support functionalmodules and pooling operations"


Differential Revision: [D16879132](https://our.internmc.facebook.com/intern/diff/D16879132/)
